### PR TITLE
revsets: allow single internal `-` and `+` characters in symbols (#46)

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

- [x] I have made relevant updates to `CHANGELOG.md`

### CODE REVIEW

File: lib/src/revset.pest [hunk 1]
```diff
@@ -12,10 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-non_period_identifier = @{ (ASCII_ALPHANUMERIC | "_" | "@" | "/")+ }
+non_final_identifier = @{ (ASCII_ALPHANUMERIC | "_" | "@" | "/")+ }
```
Comment (torquestomp) [inline lib/src/revset.pest:15]: Should this be 'non_terminal_identifier' or some other word?  'final' strictly implies that it's the end character, though I don't know that 'terminal' is much better.

File: lib/src/revset.pest [hunk 2]
```diff
@@ -12,10 +12,10 @@
...
 identifier = @{
-  (non_period_identifier+ ~ ".")+ ~ non_period_identifier+
-  | non_period_identifier+
+  (non_final_identifier+ ~ ("." | "-" | "+" ))+ ~ non_final_identifier+
```
Comment (torquestomp) [inline lib/src/revset.pest:17]: How will this handle 'foo--bar', 'foo++bar', 'foo+-bar', etc.?  I don't know that they should be rejected as valid identifiers, but if so they should be tested.

Comment (martinvonz) [reply lib/src/revset.pest]: I meant to reject them, yes. That leaves the door open for using e.g. `++` as an infix operator in the future, and it's easier to relax it later than to reverse it if we relax it now. Good point about testing, I've added a few.

File: docs/revsets.md [hunk 3]
```diff
@@ -34,9 +34,9 @@ typically only one visible commit with a given change ID). A unique prefix of
 the full change ID can also be used. It is an error to use a non-unique prefix.
 
 Use double quotes to prevent a symbol from being interpreted as an expression.
-For example, `"x-1"` is the symbol `x-1`, not the parents of symbol `x`.
+For example, `"x-"` is the symbol `x-`, not the parents of symbol `x`.
 Taking shell quoting into account, you may need to use something like
-`jj log -r '"x-1"'`.
+`jj log -r '"x-"'`.
```
Comment (torquestomp) [inline docs/revsets.md:39]: Not sure this section makes sense now, "x-" isn't actually a valid expression, is it?

**Review by torquestomp — CHANGES_REQUESTED** [review]

File: docs/revsets.md (hunk 3 above)
Comment (martinvonz) [reply docs/revsets.md:39]: It is valid - it means "parents of x" (see https://github.com/martinvonz/jj/blob/main/docs/revsets.md#operators)

File: docs/revsets.md [hunk 4]
```diff
@@ -34,9 +34,9 @@ typically only one visible commit with a given change ID). A unique prefix of
 the full change ID can also be used. It is an error to use a non-unique prefix.
```
Comment (tp-woven) [inline docs/revsets.md:34]: Should there be a mention here somewhere about adding spaces around (or parentheses around the identifier preceding) `+`/`-` to force parsing as an operator? (Assuming that's actually the case, of course...)

Comment (martinvonz) [reply docs/revsets.md:34]: Those two operators are only postfix operators right now and no space is needed.

Comment (tp-woven) [reply docs/revsets.md:34]: Ah, gotcha, thanks for clarifying that 😄

**Review by torquestomp — APPROVED** [review]

**Pull request merged**